lib/kubeclient/common.rb

```diff
@@ -126,28 +130,41 @@ def rest_client
       end
     end
 
-    def watch_entities(entity_type, resource_version = nil)
-      resource = resource_name(entity_type.to_s)
-      uri = @api_endpoint
-            .merge("#{@api_endpoint.path}/#{@api_version}/watch/#{resource}")
-
-      unless resource_version.nil?
-        uri.query = URI.encode_www_form('resourceVersion' => resource_version)
+    # Accepts the following string options:
+    #   :namespace - the namespace of the entity.
+    #   :name - the name of the entity to watch.
+    #   :label_selector - a selector to restrict the list of returned objects by their labels.
+    #   :field_selector - a selector to restrict the list of returned objects by their fields.
+    #   :resource_version - shows changes that occur after that particular version of a resource.
+    def watch_entities(entity_type, options = {})
+      ns = build_namespace_prefix(options[:namespace])
+
+      uri = @api_endpoint.dup
+      uri.path += "/#{@api_version}/watch/#{ns}#{resource_name(entity_type.to_s)}"
+      uri.path += "/#{options[:name]}" if options[:name]
```

Comment:
Minor (no need to change unless you fully agree) to be consistent with the rest of the code it should have been:

    path =  "/#{@api_version}/watch/#{ns}#{resource_name(entity_type.to_s)}"
    path += "/#{options[:name]}" if options[:name]
    uri = @api_endpoint.merge(path)